### PR TITLE
Minor refactor for readability

### PR DESCRIPTION
**Pull request by som-snytt**
Follow-up to last week's commit, to align arrows, and also use more arrows.

This demonstrates why we need `case if` syntax.

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Types.scala [hunk 1]
```diff
@@ -419,7 +419,7 @@ trait Types
      *  The empty list for all other types */
     def parents: List[Type] = List()
 
-    /** For a class with nonEmpty parents, the first parent.
+    /** For a class with !isEmpty parents, the first parent.
```
Comment (hrhino) [inline src/reflect/scala/reflect/internal/Types.scala:422]: sed, or subtle snark?

Comment (som-snytt) [reply src/reflect/scala/reflect/internal/Types.scala:422]: It wasn't meant to be subtle, but I tired of converting the other comments. Who knew nonempty was so often commented? The resultant effect is subtler than intended.

**Comment by som-snytt** [comment]
Squashed just to force the stupid travis rebuild.

**Comment by hrhino** [comment]
Travis, rebuild this!

or

@travis-ci rebuild

or, for nostalgia to before my time, can we get the Travis folks to give us PLS REBUILD again?

File: src/reflect/scala/reflect/internal/Types.scala [hunk 2]
```diff
@@ -2723,15 +2722,11 @@ trait Types
...
-          infixTypeString
-        else if (isTupleTypeDirect(this))
-          tupleTypeString
-        else if (sym.isAliasType && prefixChain.exists(_.termSymbol.isSynthetic) && (this ne dealias))
-          "" + dealias
-        else
-          ""
+        case _ if isShowAsInfixType                    => infixTypeString
+        case _ if isTupleTypeDirect(this)              => tupleTypeString
+        case _ if sym.isAliasType && prefixChain.exists(_.termSymbol.isSynthetic) && (this ne dealias)
```
Comment (diesalbla) [inline src/reflect/scala/reflect/internal/Types.scala]:
```suggestion
        case _ if sym.isAliasType && (this ne dealias) && prefixChain.exists(_.termSymbol.isSynthetic)
```

File: src/reflect/scala/reflect/internal/Types.scala [hunk 3]
```diff
@@ -3372,14 +3367,14 @@ trait Types
     }
   }
 
-  /** Precondition: params.nonEmpty.  (args.nonEmpty enforced structurally.)
+  /** Precondition: !params.isEmpty.  (args.nonEmpty enforced structurally.)
```
Comment (diesalbla) [inline src/reflect/scala/reflect/internal/Types.scala:3370]: Do we need also change in the comment? If so, ~shout~ should it be coherent?

**Review by diesalbla — APPROVED** [review]

File: src/reflect/scala/reflect/internal/Types.scala (hunk 3 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/Types.scala:3370]: I'd definitely shout it, Be coherent!

File: src/reflect/scala/reflect/internal/Types.scala (hunk 2 above)
Comment (diesalbla) [reply src/reflect/scala/reflect/internal/Types.scala]: It's not just readability, it's also bringing a quicker condition first and short-cutting the `exists`.

**Comment by diesalbla** [comment]
@som-snytt Is this ready to go?

**Comment by som-snytt** [comment]
rebuild didn't seem to reset the stuck pending status.

Once the green check clears, the minor refactor still looks minor, plus your performance suggestion, and the arrows line up, so LGTM.

**Review by lrytz — APPROVED** [review]

**Pull request merged**